controllers/database_reconciler_helpers.go

```diff
@@ -154,11 +168,15 @@ func (r *DatabaseReconciler) handleExternalDelete(ctx context.Context, database
 		}
 		// Update the CR status if the database response is empty so that it triggers a provision operation
 		if databaseResponse.Status == common.DATABASE_CR_STATUS_EMPTY {
-			log.Info("The database might have been deleted externally, setting an empty status so it can be re-provisioned.")
+			infoStatement := "The database might have been deleted externally, setting an empty status so it can be re-provisioned."
+			log.Info(infoStatement)
+			r.recorder.Event(database, "Normal", "ExternalDeleteDetected", infoStatement)
 			database.Status.Status = common.DATABASE_CR_STATUS_EMPTY
 			err = r.Status().Update(ctx, database)
 			if err != nil {
-				log.Error(err, "Failed to update database status")
+				errStatement := "Failed to update database status"
```

Comment:
while you are changing this message, can you also improve it to indicate:
"Failed to update database status to " + database.Status.Status